### DISCUSSION TITLE
Welcome to Pi-Pico-ExpressionPedal2Midi Discussions!

### DISCUSSION THREAD

**Discussion by ashstrahle** [ashstrahle/Pi-Pico-ExpressionPedal2Midi#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by cyberhippie181** [comment]
Hi, I'm Johnny. I am kinda just beginning on gh. I really like this project, and I wondered if the code  (and the pico) could support 2 expression pedals? I know this would require some additional code, could you help with this?

**Comment by ashstrahle** [comment]
Hi Johnny (@cyberhippie181), I don't see any reason why it can't be done. There's enough I/O on the board to support it. It will mean each pedal will need to be calibrated is all.

**Comment by ashstrahle** [comment]
If you can get set up with the extra hardware (resistors, jack, expression pedal) I'll spin up a branch, update the code and get you to test. Let me know if/when you're good to go

**Comment by cyberhippie181** [comment]
Ok, great! I'll order the pico, I have everything else. I was doing this on arduino, but I think the pico is a more robust platform. I'll let you know when I'm  ready. and THANKS!!

**Comment by cyberhippie181** [comment]
I got  my RPI pico. Ready when you are

**Comment by ashstrahle** [comment]
Great. I've created a branch [MultiplePedals](https://github.com/ashstrahle/Pi-Pico-ExpressionPedal2Midi/tree/MultiplePedals). Pull the code down for that, and follow the instructions for wiring, noting the following differences for the jack rings of each pedal:
Pin 31 is for Pedal 1
Pin 32 is for Pedal 2
Pin 34 is for Pedal 3

Please let me know how you go 🙂

**Reply by cyberhippie181** [reply]
Thank you for the additional code, I think I understand it a bit better now. I won't have time to try it until at least Monday. I'll keep you posted

**Comment by cyberhippie181** [comment]
I have run into some issue, not sure what it is yet. I don't have much time right now to work on it either. I'll keep you posted

**Comment by celsomachadojr** [comment]
Hi, 
the project is just what I am looking for!! I am a totally beginner when it comes to programming or electronic, but I would like to have a try. Could you check my Amazon list https://www.amazon.de/hz/wishlist/ls/2Y0Q8UI80HDP7?ref_=wl_share to confirm wether I am buying the necessary things for the project? Thank you very much!! Best Regards, Celso Machado

**Comment by ashstrahle** [comment]
Hi Celso.

Thanks for taking an interest. One item is not correct - you’re ordering 10K Ohm resistors - they need to be 10 Ohm resistors. 

You’re also buying a lot of some of these items, but I’m guessing they’re only supplied in bulk. 

Lastly, I’d also suggest getting some leads with pins for connecting everything together if you don’t already have them.

Good luck,

Ashley

**Comment by celsomachadojr** [comment]
Hi Ashley, 

Thank you so much for the promptly reply and for taking the time to check the list!

I changed the resistor from 10k Ohms to 10 Ohms  https://www.amazon.de/hz/wishlist/ls/2Y0Q8UI80HDP7?ref_=wl_share 

Yes, they are in bulk because I couldn’t buy it separately.

I think the leads with pins are the patch cables that come with the breadboard. 

Have a good weekend and again, thanks!

Best Regards

Celso

**Comment by celsomachadojr** [comment]
Hi Ashley, 
i soldered the pin headers to the raspberry pi pico (first time, what a pain ! lol) and the cables to the TRS socket and the other cables to MIDI socket. Your instructions for the MIDI Jack cabling is as follows:
"Connect a 10Ω resistor to pin 6 on the Pico (UART1 TX). T’other end of the resistor to pin 4 of your midi socket
Connect a 33Ω resistor to pin 36 on the Pico (3V3 OUT). T’other end of the resistor to pin 5 of your midi socket
Connect a Pico ground pin (any of 3, 8, 13, 18, 23, 28, or 33) to pin 2 of your midi socket"

However in the Picture, it looks like you connected the other end of the 10Ω resistor to the pin 3 of the midi socket. 
The other end of the 33Ω to the pin 4. 
the ground pin to the 2 of the midi socket, which looks as described to me. 
Can you confirm what is right? 

Hi Ashley,

I soldered the pin headers to the Raspberry Pi Pico (first time—what a pain! lol), and connected the cables to the TRS and MIDI sockets. Your MIDI Jack cabling instructions say:

"Connect a 10Ω resistor to pin 6 on the Pico (UART1 TX). T’other end of the resistor to pin 4 of your midi socket
Connect a 33Ω resistor to pin 36 on the Pico (3V3 OUT). T’other end of the resistor to pin 5 of your midi socket
Connect a Pico ground pin (any of 3, 8, 13, 18, 23, 28, or 33) to pin 2 of your midi socket"

However, in the picture, it looks like the other end of the 10Ω resistor goes to pin 3, the other end of the 33Ω to pin 4, and the ground to pin 2. Could you clarify which is correct?

Also, I installed CircuitPython 9.1.4 and added the adafruit_midi library from the 9.x bundle to the Pico’s lib folder. Is that correct?

I’m really hoping I can get this to work. I’ll be touring in Brazil for two months in November and December, and if I can finish building it in time, I’ll be able to ditch a MIDI controller from my rig. Right now, I’m only using it to connect my expression pedal to a delay pedal that accepts MIDI. I really didn´t need to do it all, but your solution is just what I need, that´s why I thought I should have a go. However, I am really dealing with it not knowing absolutely anything what I am doing hahhahaha let´s see how far it goes! ;)

Thank you so much! 
Best regards,
Celso

**Reply by ashstrahle** [reply]
Hi Celso.

This might clarify for you (see below). The picture and the instructions are correct. The pin ordering on the midi connector aren't sequential. 

The yellow wire is connected to 10Ω and to midi pin 4
The white wire is connect to 33Ω and to midi pin 5
The brown wire is connected to ground and to midi pin 2

Also, I believe what you've doing with CircuitPython to be correct, however it's been a very long time since I did it. By all means try compiling/running, you'll soon know if it's not.

Let me know how you go.

Regards,

Ashley

<img width="1051" alt="image" src="https://github.com/user-attachments/assets/03efa98d-e5cb-4eca-a7de-37ae5e4d6e8e">
